location-service/src/main/java/com/movie/recommender/location/controller/LocationController.java

```diff
@@ -46,4 +48,10 @@ public ResponseEntity<LocationDTO> updateLocationForUser(
         LocationDTO updatedLocation = locationService.updateLocationByUserId(userId, locationUpdateDTO);
         return ResponseEntity.ok(updatedLocation);
     }
+
+    @GetMapping("/countries/cities")
+    public ResponseEntity<Map<String, List<String>>> getAllCountriesAndCities() {
```

Comment:
use a dedicated DTO for the result 
this allows to encapsulate the `Map<String, List<String>>` structure in a more readable form
something like `CountryWithCitiesDTO`
place this DTO in `common` project so it can be reused by `crawler`